tests/applications/nexus_helpers_test.py

```diff
@@ -255,3 +247,152 @@ def test_nxevent_data_ev44_generator_yields_frame_by_frame() -> None:
+
+    assert () in store
+    assert len(store[()]['children']) == 2
+    values = find_nexus_structure(store[()], ('value',))
+    assert values['module'] == 'dataset'
+    times = find_nexus_structure(store[()], ('time',))
+    assert times['module'] == 'dataset'
+    assert values['config']['values'].shape[1:] == shape
+    assert values['attributes'][0]['values'] == 'km'
+
+
+@pytest.fixture()
+def nexus_template_with_streamed_tdct():
+    return {
+        "name": "chopper_3",
+        "children": [
+            {
+                "module": "tdct",
+                "config": {"source": "source", "topic": "dream_choppers"},
+            },
+        ],
+    }
+
+
+def tdct_event_generator():
+    max_last_timestamp = 0
+    counter = 0
+    while True:
+        timestamps = (
+            np.random.randint(0, 1000_000, 100, dtype='uint64').cumsum()
+            + max_last_timestamp
+        )
+        yield dict(  # noqa: C408
+            timestamps=timestamps,
+            sequence_counter=counter,
+            name="chopper_3",
+        )
+        counter += 1
+        max_last_timestamp = timestamps.max()
+
+
+def test_tdct(nexus_template_with_streamed_tdct):
+    store = {}
+    for _, data in zip(range(10), tdct_event_generator(), strict=False):
+        merge_message_into_nexus_store(
+            nexus_template_with_streamed_tdct,
+            store,
+            data,
+            "tdct",
+        )
+
+    assert ('top_dead_center',) in store
+    tdct = store[('top_dead_center',)]
+    assert tdct['module'] == 'dataset'
+    assert np.issubdtype(
+        tdct['config']['values'].dtype, np.dtype(tdct['config']['dtype'])
+    )
+
+
+@pytest.fixture()
+def nexus_template_with_mixed_streams(
+    nexus_template_with_streamed_log, nexus_template_with_streamed_tdct
+):
+    return {
+        "children": [
+            nexus_template_with_streamed_log,
+            nexus_template_with_streamed_tdct,
+        ],
+    }
+
+
+@pytest.mark.parametrize('dtype', ['uint32'])
+@pytest.mark.parametrize('shape', [(2, 1, 3)])
+def test_mixed_streams(nexus_template_with_mixed_streams, shape, dtype):
+    store = {}
+    for _, tdct, f144 in zip(
+        range(10),
+        tdct_event_generator(),
+        f144_event_generator(shape, dtype),
+        strict=False,
+    ):
+        merge_message_into_nexus_store(
+            nexus_template_with_mixed_streams,
+            store,
+            f144,
+            "f144",
+        )
+        merge_message_into_nexus_store(
+            nexus_template_with_mixed_streams,
+            store,
+            tdct,
+            "tdct",
+        )
+    result = combine_nexus_store_and_structure(nexus_template_with_mixed_streams, store)
+    assert len(result['children']) == 2
+    log = find_nexus_structure(result, ('the_log_name',))
+    assert len(log['children']) == 2
+    tdct = find_nexus_structure(
+        result,
+        (
+            'chopper_3',
+            'top_dead_center',
+        ),
+    )
```

Comment:
This one will be overwritten by the `tdct` variable of the `for` loop and not used for any tests.... right...?